### DISCUSSION TITLE
How to build the where clause depending on some conditions

### DISCUSSION THREAD

**Discussion by wenhsinhsieh** [fnc12/sqlite_orm#1119] [Q&A] [answered]
I'd like to build the where clause with some conditions. For example:

```
auto w = where();
if (something)
    w &= c(&MyStruct::aaa) >= 0;
else
    w &= c(&MyStruct::aaa) < 0;
if (something_else)
    w &= c(&MyStruct::bbb) == "bbbbbb";
// and more conditions...

//finally do the query:
data = storage.select(cols,  w);
```

I can't find a useful operator in where_t that helps in this case, and because the where clause takes template so I can't declare it outside the if-statement.
Is this achievable with sqlite_orm?

**Comment by fnc12** [comment, marked as answer]
check out this https://github.com/fnc12/sqlite_orm/issues/1093

**Reply by wenhsinhsieh** [reply]
Thanks for your reply, this works for me.
By the way the example seems wrong, it should "or" the flag and the condition together, not "and":
`where( (not betweenFlag or between(&Actions::StopTime, "strStartTime", "strStopTime")) and (not inFlag or in(&Actions::FilterNo, {1,2,3,4,5})))`

**Reply by fnc12** [reply]
are you sure? Did you make some tests?

**Reply by wenhsinhsieh** [reply]
Yes I did and using "and" didn't work for me (the query didn't hit anything regardless the value of the flag)

**Reply by fnc12** [reply]
so nice if it works for you

**Reply by wenhsinhsieh** [reply]
My understanding:

not flag or X:
when flag is true, the result is X (which means the data is selected if X is true)
when flag is false, the result is true (which means the data is always select, i.e. ignoring X)

not flag and X:
when flag is true, the result is false (which means the data is never selected???)
when flag is false, the result is X  (which means the data is selected if X is true)

Isn't it so?

**Reply by fnc12** [reply]
well boolean tables are difficult to repro in my head that's why I use truth tables in Numbers/Excel. So if it works for you it is nice
